---
title: "Merge files into phyloseq"
author: "Alba Troci"
date: "`r Sys.Date()`"
output:
   #BiocStyle::html_document:
   html_document:
     
     code_folding: hide
editor_options:
  chunk_output_type: console
params:
  FIGPATH: "figures/"
  d.out: "./"
---

# Preparations
## Set global options

```{r style, results="asis", cache=FALSE, message = F}
# Set knit global options
library("knitr")
options(digits = 2, width = 80)
golden_ratio <- (1 + sqrt(5)) / 2
opts_chunk$set(echo = TRUE,
               tidy = FALSE,
               include = TRUE,
               fig.path = params$FIGPATH,
               dev=c("png",'pdf'),
               fig.height = 5,
               fig.width = 4 * golden_ratio,
               comment = '  ',
               dpi = 300,
               cache = FALSE)


# Pretty outputs
library("rmarkdown")
#library("BiocStyle")
library("ggthemes") # add ggplot2 themes
library("ggplot2")
library("ggpubr")

# Set seed for reproducibility
set.seed(100)

# Set plotting theme
theme_set(theme_few(base_size = 14))

# Set output directory
d.out <- params$d.out
rm(params)
```


## Load libraries for the session
```{r libraries, message=FALSE}
library("magrittr")
library("ggplot2")
library("tidyverse")
library("readxl")
#library("dada2")
library("phyloseq")
library("gridExtra")
library("vegan")
library("plyr")
library("scales")
library("reshape2")
#library("DESeq2")
library("dplyr") #for data manipulation
#library("msa") #multiple sequence alignment
library("ape") #to create phylogenetic tree
library("randomForest") 
library("caret") 
library("broom")
library("mlbench")
library("plotROC")

```


# Import data

Data from DADA2 algorithm (tax table and otu table)

```{r import data}
#ASV table
library(magrittr) 
library(dplyr)
d.micro.new <- "~/Documents/Daten /Dada2/Seq_data"
micro.new <- "/seqtab_nochim.Rds" %>% 
  paste0(d.micro.new,"/", .) %>% 
  readRDS() %>%
  data.frame()

```

```{r}

#Tax table

tax <- read.table(file = '~/Documents/Daten /Dada2/Seq_data/tables/taxa_Bayesian_RDP16_SV.tsv', sep = '\t', header = TRUE)


```


```{r}
#import MS table
library(readxl)
Datensatz <- read_excel("~/Documents/dietproject/Microbiom/metadata.edit.xlsx")

pd1 <- as.data.frame(Datensatz)

```

edit metatable

```{r}
library(readxl)
edit_seq_table <- read_excel("~/Documents/dietproject/Microbiom/edit.seq.table.xlsx")

```


```{r}
edit_seq_table$...2 <- NULL
edit_seq_table$`NGS-run` <- NULL
```

```{r}
df <- merge(edit_seq_table, pd1, by="ID") 

```

```{r}
df1 <- na.omit(df)
```


##add dietary data into the metadata table

```{r}
library(readxl)
diet <- read_excel("~/Documents/dietproject/Microbiom/dietary.data.xlsx")

```


####merge df1 and diet table

```{r}

df2 <- merge(df1, diet, by = "ID")

```




###edit otu table

```{r}
micro.new2 <- micro.new[(rownames(micro.new) %in% df2$NGS_ID), ]
```

```{r}
df3 <- df2[(df2$NGS_ID %in% rownames(micro.new2)), ]
```


sort sample names in metadata table in the same order as in otu table

```{r}
bb <- data.frame(NGS_ID=rownames(micro.new2))
bb$nr <- rownames(bb)
bb$nr <- as.numeric(as.character(bb$nr))

```

```{r}
df4 <- merge(df3, bb, by="NGS_ID")
```

```{r}
df5 <- df4[!(df4$NGS_ID %in% "23Jul1363-DL060"), ]
```

```{r}
micro.new3 <- micro.new2[!(rownames(micro.new2) %in% "23Jul1363-DL060"), ]
```

##make the otu table in the same ID order as metadata table

```{r}
df6 <- df5[order(df5$nr),] 
rownames(df6) <- df6$NGS_ID
```



```{r}
head(rownames(df6))
head(rownames(micro.new3))

identical(rownames(df6), rownames(micro.new3))
```

```{r}
library("tidyft")
tax1 <- data.table(tax)

tax2 <- separate(tax1, Kingdom.Phylum.Class.Order.Family.Genus.Species, 
                 into=c("ASV", "Kingdom", "Phylum", "Class", "Order","Family","Genus","Species"), sep = " ", remove = TRUE)

```

##edit tax table

```{r}
library("tidyverse")
tax2$ASV2 <- tax2$ASV
tax3 <- tax2 %>% remove_rownames %>% column_to_rownames(var="ASV")
```

```{r}
tax4 <- as.matrix(tax3)
```



# Phyloseq: Combine Data

Build separate components for phyloseq

```{r}
library("phyloseq")
tax.new <- tax_table(tax4)
micro.new3 <- otu_table(micro.new3, taxa_are_rows = FALSE)
df6 <- sample_data(df6)


head(taxa_names(tax.new))
head(taxa_names(micro.new3))
head(sample_names(df6))

```

Combine all data tables into a single object

```{r}
ps <- phyloseq(tax_table(tax.new), 
               sample_data(df6), 
               otu_table(micro.new3, taxa_are_rows = FALSE))
ps

```





# Save output
```{r}
"ps.diet.rds" %>%
  saveRDS(ps, .)

```


```{r}
# Konvertiere df6 aus dem phyloseq-Objekt in einen DataFrame und entferne eventuelle Attributspezifikationen
df6 <- as(sample_data(ps), "data.frame")
# Berechne den Prozentsatz der `NA`-Werte in der Korrelationsmatrix

# Extrahiere OTU-Tabelle und Taxonomiedaten als DataFrames
otu_data <- as.data.frame(otu_table(ps))
tax_data <- as.data.frame(tax_table(ps))

# Finde die Schnittmenge der Zeilennamen beider Tabellen
common_otus <- intersect(rownames(otu_data), rownames(tax_data))

# Filtere sowohl OTU- als auch Taxonomiedaten auf die gemeinsamen OTUs
otu_data <- otu_data[common_otus, ]
tax_data <- tax_data[common_otus, ]

# Füge die Genus-Spalte zur OTU-Tabelle hinzu
otu_data$Genus <- tax_data$Genus

otu_data <- as.data.frame(otu_table(ps))
colnames(otu_data) <- substr(unlist(colnames(otu_data)), 1, 10)
# Aggregiere die Daten nach Gattung und berechne die Summe der Abundanzen
otu_genus <- otu_data %>%
  group_by(Genus) %>%
  summarise(across(everything(), ~sum(.x, na.rm = TRUE)))

# Überprüfe die Anzahl der eindeutigen Gattungen und wähle so viele wie möglich (maximal 50)
num_genera <- min(50, nrow(otu_genus))
set.seed(100)  # Für Reproduzierbarkeit
otu_genus_subset <- otu_genus %>% sample_n(num_genera)

# Entferne die Genus-Spalte und wandle sie in eine Matrix um
rownames(otu_genus_subset) <- otu_genus_subset$Genus
otu_genus_subset$Genus <- NULL
otu_genus_subset <- as.matrix(otu_genus_subset)

# Wähle die Nährstoff-Daten (Beispiel: Protein und Fett) aus dem DataFrame
nutrient_data <- df6 %>% select(Protein, Fat)

# **Synchronisiere die Proben (Zeilen) in den Abundanz- und Nährstoffdaten**

# Finde die Schnittmenge der Proben zwischen beiden Datensätzen
common_samples <- intersect(rownames(otu_genus_subset), rownames(nutrient_data))

# Filtere beide Datensätze auf die gemeinsamen Proben
otu_genus_subset <- otu_genus_subset[common_samples, ]
nutrient_data <- nutrient_data[common_samples, ]

# Wende eine Log-Transformation auf die OTU-Daten an
otu_genus_subset <- log1p(otu_genus_subset)

# Berechne die Spearman-Korrelationsmatrix
correlation_matrix <- cor(otu_genus_subset, nutrient_data, method = "spearman")

# Visualisiere die Korrelationsmatrix
library(ggcorrplot)
ggcorrplot(correlation_matrix, method = "square", type = "lower", 
           lab = TRUE, title = "Optimized Spearman Correlation: Bacteria vs Nutrients", 
           colors = c("blue", "white", "red"))

```



```

