---
title: "Merge files into phyloseq"
author: "Mona Hildebrandt"
date: "`r Sys.Date()`"
output:
   #BiocStyle::html_document:
   html_document:
     
     code_folding: hide
editor_options: 
  chunk_output_type: console
params:
  FIGPATH: "figures/"
  d.out: "./"
---

# Preparations
## Set global options

```{r style, results="asis", cache=FALSE, message = F}
# Set knit global options
library("knitr")
options(digits = 2, width = 80)
golden_ratio <- (1 + sqrt(5)) / 2
opts_chunk$set(echo = TRUE,
               tidy = FALSE,
               include = TRUE,
               fig.path = params$FIGPATH,
               dev=c("png",'pdf'),
               fig.height = 5,
               fig.width = 4 * golden_ratio,
               comment = '  ',
               dpi = 300,
               cache = FALSE)


# Pretty outputs
library("rmarkdown")
#library("BiocStyle")
library("ggthemes") # add ggplot2 themes
library("ggplot2")
library("ggpubr")

# Set seed for reproducibility
set.seed(100)

# Set plotting theme
theme_set(theme_few(base_size = 14))

# Set output directory
d.out <- params$d.out
rm(params)
```


## Load libraries for the session
```{r libraries, message=FALSE}
library("magrittr")
library("ggplot2")
library("tidyverse")
library("readxl")
#library("dada2")
library("phyloseq")
library("gridExtra")
library("vegan")
library("plyr")
library("scales")
library("reshape2")
#library("DESeq2")
library("dplyr") #for data manipulation
#library("msa") #multiple sequence alignment
library("ape") #to create phylogenetic tree
library("randomForest") 
library("caret") 
library("broom")
library("mlbench")
library("plotROC")

```


# Alpha diversity {.tabset}

## No Filter taxa

### Import data


Filtered phyloseq, stool samples
I have filtered here only ASV with prevalence > 5% of all samples
```{r}
# Phyloseq object AD filtered samples
d.ps.filter.ad.st <- "~/Documents/dietproject"
ps.filter.ad.st <- "/3.ps.filter.pd.abundance.rds" %>%
  paste0(d.ps.filter.ad.st,"/", .) %>% 
  readRDS() 
ps.filter.ad.st


```

subset only stool samples:


```{r}
ps.st <- ps.filter.ad.st
ps.st
```

```{r}
ps.st <- subset_samples(ps.filter.ad.st, !(sample_names(ps.filter.ad.st) %in% "23Jul94-DL185"))
ps.st
```



### Rarefaction

I rarefy samples before alpha diversity
I will subsample to the min number of reads

```{r}
ps.rar <- rarefy_even_depth(ps.st, sample.size = min(sample_sums(ps.st)))
ps.rar

```


### Calculate alpha diversity and plot 


Calculate alpha diversity measures
```{r}
#Diversity values in one table only
rich <- estimate_richness(ps.rar, measures = c("Shannon", "Chao1"))
head(rich)

```

Prepare table: merge diversity measure results with metadata
```{r}
df <- cbind(sample_data(ps.rar), rich)
```

Lets have a look on the metadata table

```{r}
meta <- sample_data(ps.rar)
```


### Plot

Edit: transform  tables to fit ggplot

```{r}
df55 <- df[c("Chao1", "Protein.in..", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df55$Value <- df55$Chao1
df55$Measure <- "Chao1"
df55$Chao1 <- NULL


df66 <- df[c("Shannon", "Protein.in..", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df66$Value <- df66$Shannon
df66$Measure <- "Shannon"
df66$Shannon <- NULL
```

```{r}
colnames(df55) <- c("Protein", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```

```{r}
colnames(df66) <- c("Protein", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```


```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df55, aes(x=Protein, y=Value, color=Protein)) + 
     geom_point(na.rm = TRUE, color="darkblue", size=2, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "red3")+
                        theme(text = element_text(size = 17)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Chao1)", x = "Protein")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
#ggsave("alpha.protein.png")
```

```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df66, aes(x=Protein, y=Value, color=Protein)) + 
     geom_point(na.rm = TRUE, color="darkblue", size=4, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "red3", linewidth = 2)+
                        theme(text = element_text(size = 40)) +
                       # scale_colour_brewer(palette = "pink") +
                        labs(y="Diversity richness (Shannon)", x = "Protein (%)")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
ggsave("shannon.protein.png")
```

##Carbs 
```{r}
df77 <- df[c("Chao1", "Kohlenhydrate.in..", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df77$Value <- df77$Chao1
df77$Measure <- "Chao1"
df77$Chao1 <- NULL


df88 <- df[c("Shannon", "Kohlenhydrate.in..", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df88$Value <- df88$Shannon
df88$Measure <- "Shannon"
df88$Shannon <- NULL
```

```{r}
colnames(df77) <- c("Kohlenhydrate", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```

```{r}
colnames(df88) <- c("Kohlenhydrate", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```


```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df77, aes(x=Kohlenhydrate, y=Value, color=Kohlenhydrate)) + 
     geom_point(na.rm = TRUE, color="darkblue", size=1.5, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "red3")+
                        theme(text = element_text(size = 13)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Chao1)", x = "Carbohydrate")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
#ggsave("alpha.kohlenhyrate.png")
```

```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df88, aes(x=Kohlenhydrate, y=Value, color=Kohlenhydrate)) + 
     geom_point(na.rm = TRUE, color="darkblue", size=4, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "red3", linewidth = 2)+
                        theme(text = element_text(size = 40)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Shannon)", x = "Carbohydrate (%)")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
ggsave("shannon.carbohydrate.png")
```

##fat
```{r}
df99 <- df[c("Chao1", "Fett.in..", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df99$Value <- df99$Chao1
df99$Measure <- "Chao1"
df99$Chao1 <- NULL


df100<- df[c("Shannon", "Fett.in..", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df100$Value <- df100$Shannon
df100$Measure <- "Shannon"
df100$Shannon <- NULL
```

```{r}
colnames(df99) <- c("Fett", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```

```{r}
colnames(df100) <- c("Fett", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```


```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df99, aes(x=Fett, y=Value, color=Fett)) + 
     geom_point(na.rm = TRUE, color="darkblue", size=1.5, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "red3")+
                        theme(text = element_text(size = 13)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Chao1)", x = "Fat")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
#ggsave("alpha.fett.png")
```

```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df100, aes(x=Fett, y=Value, color=Fett)) + 
     geom_point(na.rm = TRUE, color="darkblue", size=4, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "green4", linewidth = 2)+
                        theme(text = element_text(size = 40)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Shannon)", x = "Fat (%)")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
ggsave("shannon.fat.png")
```


##fat_PUFA
```{r}
df101 <- df[c("Chao1", "Fett_MUFS.in..", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df101$Value <- df101$Chao1
df101$Measure <- "Chao1"
df101$Chao1 <- NULL


df102<- df[c("Shannon", "Fett_MUFS.in..", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df102$Value <- df102$Shannon
df102$Measure <- "Shannon"
df102$Shannon <- NULL
```

```{r}
colnames(df101) <- c("Fett_MUFA", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```

```{r}
colnames(df102) <- c("Fett_MUFA", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```


```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df101, aes(x=Fett_MUFA, y=Value, color=Fett_MUFA)) + 
     geom_point(na.rm = TRUE, color="darkblue", size=1.5, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "red3")+
                        theme(text = element_text(size = 13)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Chao1)", x = "Fat_PUFA")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
#ggsave("alpha.Fett_MUFA.png")
```

```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df102, aes(x=Fett_MUFA, y=Value, color=Fett_MUFA)) + 
     geom_point(na.rm = TRUE, color="darkblue", size=4, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "green4", linewidth = 2)+
                        theme(text = element_text(size = 40)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Shannon)", x = "PUFA (%)")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
ggsave("shannon.PUFA.png")
```


##sugar
```{r}
df103 <- df[c("Chao1", "davon_Zucker.in..", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df103$Value <- df103$Chao1
df103$Measure <- "Chao1"
df103$Chao1 <- NULL


df104<- df[c("Shannon", "davon_Zucker.in..", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df104$Value <- df104$Shannon
df104$Measure <- "Shannon"
df104$Shannon <- NULL
```

```{r}
colnames(df103) <- c("Zucker", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```

```{r}
colnames(df104) <- c("Zucker", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```


```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df103, aes(x=Zucker, y=Value, color=Zucker)) + 
     geom_point(na.rm = TRUE, color="darkblue", size=1.5, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "red3")+
                        theme(text = element_text(size = 13)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Chao1)", x = "Sugar")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
#ggsave("alpha.sugar.png")
```

```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df104, aes(x=Zucker, y=Value, color=Zucker)) + 
     geom_point(na.rm = TRUE, color="darkblue", size=4, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "red3", linewidth = 2)+
                        theme(text = element_text(size = 40)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Shannon)", x = "Sugar (%)")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
ggsave("shannon.sugar.png")
```

##fiber
```{r}
df105 <- df[c("Chao1", "Ballaststoffe.in..", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df105$Value <- df105$Chao1
df105$Measure <- "Chao1"
df105$Chao1 <- NULL


df106<- df[c("Shannon", "Ballaststoffe.in..", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df106$Value <- df106$Shannon
df106$Measure <- "Shannon"
df106$Shannon <- NULL
```

```{r}
colnames(df105) <- c("Ballaststoffe", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```

```{r}
colnames(df106) <- c("Ballaststoffe", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```


```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df105, aes(x=Ballaststoffe, y=Value, color=Ballaststoffe)) + 
     geom_point(na.rm = TRUE, color="darkblue", size=2, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "red3")+
                        theme(text = element_text(size = 17)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Chao1)", x = "Fiber")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
#ggsave("alpha.fiber.png")
```

```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df106, aes(x=Ballaststoffe, y=Value, color=Ballaststoffe)) + 
     geom_point(na.rm = TRUE, color="darkblue", size=4, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "red3", linewidth = 2)+
                        theme(text = element_text(size = 40)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Shannon)", x = "Fiber (%)")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
ggsave("shannon.fiber.png")
```

#Micronutrients


#Cholsterol
```{r}
df107 <- df[c("Chao1", "Cholesterin", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df107$Value <- df107$Chao1
df107$Measure <- "Chao1"
df107$Chao1 <- NULL


df108<- df[c("Shannon", "Cholesterin", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df108$Value <- df108$Shannon
df108$Measure <- "Shannon"
df108$Shannon <- NULL
```

```{r}
colnames(df107) <- c("Cholesterin", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```

```{r}
colnames(df108) <- c("Cholesterin", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```


```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df107, aes(x=Cholesterin, y=Value, color=Cholesterin)) + 
     geom_point(na.rm = TRUE, color="purple3", size=4, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "red3", linewidth = 2)+
                        theme(text = element_text(size = 36)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Chao1)", x = "Cholesterol")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
#ggsave("alpha.fiber.png")
```

```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df108, aes(x=Cholesterin, y=Value, color=Cholesterin)) + 
     geom_point(na.rm = TRUE, color="purple3", size=4, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "red3", linewidth = 2)+
                        theme(text = element_text(size = 40)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Shannon)", x = "Cholesterol (%)")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
ggsave("shannon.cholesterol.png")
```

#Sodium
```{r}
df109 <- df[c("Chao1", "Natrium", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df109$Value <- df109$Chao1
df109$Measure <- "Chao1"
df109$Chao1 <- NULL


df110<- df[c("Shannon", "Natrium", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df110$Value <- df110$Shannon
df110$Measure <- "Shannon"
df110$Shannon <- NULL
```

```{r}
colnames(df109) <- c("Natrium", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```

```{r}
colnames(df110) <- c("Natrium", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```


```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df109, aes(x=Natrium, y=Value, color=Natrium)) + 
     geom_point(na.rm = TRUE, color="purple3", size=2, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "red3")+
                        theme(text = element_text(size = 40)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Chao1)", x = "Sodium")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
#ggsave("alpha.fiber.png")
```

```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df110, aes(x=Natrium, y=Value, color=Natrium)) + 
     geom_point(na.rm = TRUE, color="purple3", size=4, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "red3", linewidth = 2)+
                        theme(text = element_text(size = 40)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Shannon)", x = "Sodium (%)")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
ggsave("shannon.sodium.png")
```


#Potassium
```{r}
df111 <- df[c("Chao1", "Kalium", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df111$Value <- df111$Chao1
df111$Measure <- "Chao1"
df111$Chao1 <- NULL


df112<- df[c("Shannon", "Kalium", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df112$Value <- df112$Shannon
df112$Measure <- "Shannon"
df112$Shannon <- NULL
```

```{r}
colnames(df111) <- c("Kalium", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```

```{r}
colnames(df112) <- c("Kalium", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```


```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df111, aes(x=Kalium, y=Value, color=Kalium)) + 
     geom_point(na.rm = TRUE, color="purple3", size=4, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "red3")+
                        theme(text = element_text(size = 17)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Chao1)", x = "Potassium")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
#ggsave("alpha.fiber.png")
```

```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df112, aes(x=Kalium, y=Value, color=Kalium)) + 
     geom_point(na.rm = TRUE, color="purple3", size=4, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "red3", linewidth = 2)+
                        theme(text = element_text(size = 40)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Shannon)", x = "Potassium (%)")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
ggsave("shannon.potassium.png")
```


#Calcium
```{r}
df113 <- df[c("Chao1", "Calcium", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df113$Value <- df113$Chao1
df113$Measure <- "Chao1"
df113$Chao1 <- NULL


df114<- df[c("Shannon", "Calcium", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df114$Value <- df114$Shannon
df114$Measure <- "Shannon"
df114$Shannon <- NULL
```

```{r}
colnames(df113) <- c("Calcium", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```

```{r}
colnames(df114) <- c("Calcium", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```


```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df113, aes(x=Calcium, y=Value, color=Calcium)) + 
     geom_point(na.rm = TRUE, color="purple3", size=1.5, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "green4", linewidth = 2)+
                        theme(text = element_text(size = 36)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Chao1)", x = "Calcium")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
#ggsave("alpha.fiber.png")
```

```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df114, aes(x=Calcium, y=Value, color=Calcium)) + 
     geom_point(na.rm = TRUE, color="purple3", size=4, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "green4", linewidth = 2)+
                        theme(text = element_text(size = 40)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Shannon)", x = "Calcium (%)")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
ggsave("shannon.calcium.png")
```


#Phosphorus
```{r}
df115 <- df[c("Chao1", "Phosphor", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df115$Value <- df115$Chao1
df115$Measure <- "Chao1"
df115$Chao1 <- NULL


df116<- df[c("Shannon", "Phosphor", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df116$Value <- df116$Shannon
df116$Measure <- "Shannon"
df116$Shannon <- NULL
```

```{r}
colnames(df115) <- c("Phosphor", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```

```{r}
colnames(df116) <- c("Phosphor", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```


```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df115, aes(x=Phosphor, y=Value, color=Phosphor)) + 
     geom_point(na.rm = TRUE, color="purple3", size=3, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "red3", linewidth = 2)+
                        theme(text = element_text(size = 36)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Chao1)", x = "Phosphorus")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
#ggsave("alpha.fiber.png")
```

```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df116, aes(x=Phosphor, y=Value, color=Phosphor)) + 
     geom_point(na.rm = TRUE, color="purple3", size=4, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "red3", linewidth = 2)+
                        theme(text = element_text(size = 40)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Shannon)", x = "Phosphorus (%)")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
ggsave("shannon.phosphorus.png")
```


#Magnesium
```{r}
df117 <- df[c("Chao1", "Magnesium", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df117$Value <- df117$Chao1
df117$Measure <- "Chao1"
df117$Chao1 <- NULL


df118<- df[c("Shannon", "Magnesium", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df118$Value <- df118$Shannon
df118$Measure <- "Shannon"
df118$Shannon <- NULL
```

```{r}
colnames(df117) <- c("Magnesium", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```

```{r}
colnames(df118) <- c("Magnesium", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```


```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df117, aes(x=Magnesium, y=Value, color=Magnesium)) + 
     geom_point(na.rm = TRUE, color="purple3", size=1.3, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "red3")+
                        theme(text = element_text(size = 13)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Chao1)", x = "Magnesium")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
#ggsave("alpha.fiber.png")
```

```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df118, aes(x=Magnesium, y=Value, color=Magnesium)) + 
     geom_point(na.rm = TRUE, color="purple3", size=4, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "green4", linewidth = 2)+
                        theme(text = element_text(size = 40)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Shannon)", x = "Magnesium (%)")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
ggsave("shannon.magnesium.png")
```

#Iron 
```{r}
df119 <- df[c("Chao1", "Eisen", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df119$Value <- df119$Chao1
df119$Measure <- "Chao1"
df119$Chao1 <- NULL


df120<- df[c("Shannon", "Eisen", "BMI", "Geschlecht", "Alter", "Rauchen.")]
df120$Value <- df120$Shannon
df120$Measure <- "Shannon"
df120$Shannon <- NULL
```

```{r}
colnames(df119) <- c("Eisen", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```

```{r}
colnames(df120) <- c("Eisen", "BMI", "Geschlecht", "Alter", "Rauchen.", "Value" ,"Measure")
```


```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df119, aes(x=Eisen, y=Value, color=Eisen)) + 
     geom_point(na.rm = TRUE, color="purple3", size=3, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "green4", linewidth = 2)+
                        theme(text = element_text(size = 36)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Chao1)", x = "Iron")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
#ggsave("alpha.fiber.png")
```

```{r, out.width="60%"}
par(mfrow=c(1,1))
theme_set(theme_bw())
p <- ggplot(df120, aes(x=Eisen, y=Value, color=Eisen)) + 
     geom_point(na.rm = TRUE, color="purple3", size=3, pch=18)+
     geom_smooth(method=lm, se=FALSE, fullrange=TRUE, color = "green4", linewidth = 2)+
                        theme(text = element_text(size = 40)) +
                        #scale_colour_brewer(palette = "red") +
                        labs(y="Diversity richness (Shannon)", x = "Iron (%)")+
  scale_y_continuous(expand = expansion(mult = c(0, 0.3)))
 
p
  
  
ggsave("shannon.iron.png")
```



### test regression 
# Lineares Modell für Chao1 und Protein

#Protein
```{r}
lm_chao1_protein <- lm(Value ~ Protein + BMI + Geschlecht + Alter + Rauchen., data = df55)
summary(lm_chao1_protein)
```

```{r}
lm_shannon_protein <- lm(Value ~ Protein + BMI + Geschlecht + Alter + Rauchen., data = df66)
summary(lm_shannon_protein)
```

#Carbohydrate
```{r}
lm_chao1_carbohydrate <- lm(Value ~ Kohlenhydrate + BMI + Geschlecht + Alter + Rauchen., data = df77)
summary(lm_chao1_carbohydrate)
```

```{r}
lm_shannon_carbohydrate <- lm(Value ~ Kohlenhydrate + BMI + Geschlecht + Alter + Rauchen., data = df88)
summary(lm_shannon_carbohydrate)
```

#Fat 
```{r}
lm_chao1_fat <- lm(Value ~ Fett + BMI + Geschlecht + Alter + Rauchen., data = df99)
summary(lm_chao1_fat)
```

```{r}
lm_shannon_fat <- lm(Value ~ Fett + BMI + Geschlecht + Alter + Rauchen., data = df100)
summary(lm_shannon_fat)
```

#Fat_MUFA
```{r}
lm_chao1_fat_MUFA <- lm(Value ~ Fett_MUFA + BMI + Geschlecht + Alter + Rauchen., data = df101)
summary(lm_chao1_fat_MUFA)
```

```{r}
lm_shannon_fat_MUFA <- lm(Value ~ Fett_MUFA + BMI + Geschlecht + Alter + Rauchen., data = df102)
summary(lm_shannon_fat_MUFA)
```

#sugar
```{r}
lm_chao1_sugar <- lm(Value ~ Zucker + BMI + Geschlecht + Alter + Rauchen., data = df103)
summary(lm_chao1_sugar)
```

```{r}
lm_shannon_sugar <- lm(Value ~ Zucker + BMI + Geschlecht + Alter + Rauchen., data = df104)
summary(lm_shannon_sugar)
```

#fiber
```{r}
lm_chao1_fiber <- lm(Value ~ Ballaststoffe + BMI + Geschlecht + Alter + Rauchen., data = df105)
summary(lm_chao1_fiber)
```

```{r}
lm_shannon_fiber <- lm(Value ~ Ballaststoffe + BMI + Geschlecht + Alter + Rauchen., data = df106)
summary(lm_shannon_fiber)
```

#Cholesterol
```{r}
lm_chao1_cholesterol <- lm(Value ~ Cholesterin + BMI + Geschlecht + Alter + Rauchen., data = df107)
summary(lm_chao1_cholesterol)
```

```{r}
lm_shannon_cholesterol <- lm(Value ~ Cholesterin + BMI + Geschlecht + Alter + Rauchen., data = df108)
summary(lm_shannon_cholesterol)
```

#Sodium
```{r}
lm_chao1_sodium <- lm(Value ~ Natrium + BMI + Geschlecht + Alter + Rauchen., data = df109)
summary(lm_chao1_sodium)
```

```{r}
lm_shannon_sodium <- lm(Value ~ Natrium + BMI + Geschlecht + Alter + Rauchen., data = df110)
summary(lm_shannon_sodium)
```

#Potassium
```{r}
lm_chao1_potassium <- lm(Value ~ Kalium + BMI + Geschlecht + Alter + Rauchen., data = df111)
summary(lm_chao1_potassium)
```

```{r}
lm_shannon_potassium <- lm(Value ~ Kalium + BMI + Geschlecht + Alter + Rauchen., data = df112)
summary(lm_shannon_potassium)
```

#Calcium 
```{r}
lm_chao1_calcium <- lm(Value ~ Calcium + BMI + Geschlecht + Alter + Rauchen., data = df113)
summary(lm_chao1_calcium)
```

```{r}
lm_shannon_calcium <- lm(Value ~ Calcium + BMI + Geschlecht + Alter + Rauchen., data = df114)
summary(lm_shannon_calcium)
```

#Phosphorus
```{r}
lm_chao1_phosphorus <- lm(Value ~ Phosphor + BMI + Geschlecht + Alter + Rauchen., data = df115)
summary(lm_chao1_phosphorus)
```

```{r}
lm_shannon_phosphorus <- lm(Value ~ Phosphor + BMI + Geschlecht + Alter + Rauchen., data = df116)
summary(lm_shannon_phosphorus)
```

#Magsnesium 
```{r}
lm_chao1_magnesium <- lm(Value ~ Magnesium + BMI + Geschlecht + Alter + Rauchen., data = df117)
summary(lm_chao1_magnesium)
```

```{r}
lm_shannon_magnesium <- lm(Value ~ Magnesium + BMI + Geschlecht + Alter + Rauchen., data = df118)
summary(lm_shannon_magnesium)
```
#Eisen 
```{r}
lm_chao1_iron <- lm(Value ~ Eisen + BMI + Geschlecht + Alter + Rauchen., data = df119)
summary(lm_chao1_iron)
```

```{r}
lm_shannon_iron<- lm(Value ~ Eisen + BMI + Geschlecht + Alter + Rauchen., data = df120)
summary(lm_shannon_iron)
```



```{r}
# Liste der Nährstoffe und zugehörige Modelle
nutrient_models <- list(
  Protein = list(chao1 = summary(lm_chao1_protein), shannon = summary(lm_shannon_protein)),
  Kohlenhydrate = list(chao1 = summary(lm_chao1_carbohydrate), shannon = summary(lm_shannon_carbohydrate)),
  Fett = list(chao1 = summary(lm_chao1_fat), shannon = summary(lm_shannon_fat)),
  Fett_MUFA = list(chao1 = summary(lm_chao1_fat_MUFA), shannon = summary(lm_shannon_fat_MUFA)),
  Zucker = list(chao1 = summary(lm_chao1_sugar), shannon = summary(lm_shannon_sugar)),
  Ballaststoffe = list(chao1 = summary(lm_chao1_fiber), shannon = summary(lm_shannon_fiber)),
  Cholesterin = list(chao1 = summary(lm_chao1_cholesterol), shannon = summary(lm_shannon_cholesterol)),
  Natrium = list(chao1 = summary(lm_chao1_sodium), shannon = summary(lm_shannon_sodium)),
  Kalium = list(chao1 = summary(lm_chao1_potassium), shannon = summary(lm_shannon_potassium)),
  Calcium = list(chao1 = summary(lm_chao1_calcium), shannon = summary(lm_shannon_calcium)),
  Phosphor = list(chao1 = summary(lm_chao1_phosphorus), shannon = summary(lm_shannon_phosphorus)),
  Magnesium = list(chao1 = summary(lm_chao1_magnesium), shannon = summary(lm_shannon_magnesium)),
  Eisen = list(chao1 = summary(lm_chao1_iron), shannon = summary(lm_shannon_iron))
)

# Leere Liste zur Speicherung der Ergebnisse
results_list <- list()

# Schleife zur Verarbeitung der Ergebnisse
for (nutrient in names(nutrient_models)) {
  # Extrahiere die Modelle
  chao1_model <- nutrient_models[[nutrient]]$chao1
  shannon_model <- nutrient_models[[nutrient]]$shannon
  
  # Extrahiere Ergebnisse für Chao1
  terms_chao1 <- rownames(chao1_model$coefficients)
  t_chao1 <- data.frame(
    Metric = "Chao1",
    Nutrient = nutrient,
    Term = terms_chao1,
    Estimate = chao1_model$coefficients[, "Estimate"],
    StdError = chao1_model$coefficients[, "Std. Error"],
    tValue = chao1_model$coefficients[, "t value"],
    Pvalue = chao1_model$coefficients[, "Pr(>|t|)"]
  )
  
  # Extrahiere Ergebnisse für Shannon
  terms_shannon <- rownames(shannon_model$coefficients)
  t_shannon <- data.frame(
    Metric = "Shannon",
    Nutrient = nutrient,
    Term = terms_shannon,
    Estimate = shannon_model$coefficients[, "Estimate"],
    StdError = shannon_model$coefficients[, "Std. Error"],
    tValue = shannon_model$coefficients[, "t value"],
    Pvalue = shannon_model$coefficients[, "Pr(>|t|)"]
  )
  
  # Kombiniere die Ergebnisse
  results_list[[nutrient]] <- rbind(t_chao1, t_shannon)
}

# Ergebnisse in eine Tabelle zusammenführen
final_table <- do.call(rbind, results_list)

# P-Werte anpassen (Benjamini-Hochberg)
final_table$Pvalue_adj <- p.adjust(final_table$Pvalue, method = "BH")

# Sternchen für P-Werte hinzufügen
final_table$Pvalue <- paste0(
  round(final_table$Pvalue, 4),
  ifelse(final_table$Pvalue < 0.001, "***",
         ifelse(final_table$Pvalue < 0.01, "**",
                ifelse(final_table$Pvalue < 0.05, "*", "")))
)

# Sternchen für P-Werte angepasst
final_table$Pvalue_adj <- paste0(
  round(as.numeric(gsub("\\*", "", final_table$Pvalue_adj)), 4),
  ifelse(as.numeric(gsub("\\*", "", final_table$Pvalue_adj)) < 0.001, "***",
         ifelse(as.numeric(gsub("\\*", "", final_table$Pvalue_adj)) < 0.01, "**",
                ifelse(as.numeric(gsub("\\*", "", final_table$Pvalue_adj)) < 0.05, "*", "")))
)

# Ergebnisse anzeigen
final_table

```


```{r}
library(openxlsx)
write.xlsx(final_table, "6.aplpha.results.xlsx")
```



```{r}
# Liste der Nährstoffe und zugehörige Datensätze
nutrient_data <- list(
  Protein = list(data_chao1 = df55, data_shannon = df66),
  Kohlenhydrate = list(data_chao1 = df77, data_shannon = df88),
  Fett = list(data_chao1 = df99, data_shannon = df100),
  Fett_MUFA = list(data_chao1 = df101, data_shannon = df102),
  Zucker = list(data_chao1 = df103, data_shannon = df104),
  Ballaststoffe = list(data_chao1 = df105, data_shannon = df106),
  Cholesterin = list(data_chao1 = df107, data_shannon = df108),
  Natrium = list(data_chao1 = df109, data_shannon = df110),
  Kalium = list(data_chao1 = df111, data_shannon = df112),
  Calcium = list(data_chao1 = df113, data_shannon = df114),
  Phosphor = list(data_chao1 = df115, data_shannon = df116),
  Magnesium = list(data_chao1 = df117, data_shannon = df118),
  Eisen = list(data_chao1 = df119, data_shannon = df120)
)

# Liste zur Speicherung der Ergebnisse
results_list <- list()

# Schleife über alle Nährstoffe
for (nutrient in names(nutrient_data)) {
  # Hole die Datensätze für Chao1 und Shannon
  data_chao1 <- nutrient_data[[nutrient]]$data_chao1
  data_shannon <- nutrient_data[[nutrient]]$data_shannon
  
  # Chao1-Modell
  lm_chao1 <- lm(as.formula(paste("Value ~", nutrient, "+ BMI + Geschlecht + Alter + Rauchen.")), data = data_chao1)
  summary_chao1 <- summary(lm_chao1)
  
  # Shannon-Modell
  lm_shannon <- lm(as.formula(paste("Value ~", nutrient, "+ BMI + Geschlecht + Alter + Rauchen.")), data = data_shannon)
  summary_shannon <- summary(lm_shannon)
  
  # Ergebnisse für Chao1
  terms_chao1 <- rownames(summary_chao1$coefficients)
  t_chao1 <- data.frame(
    Metric = "Chao1",
    Nutrient = nutrient,
    Term = terms_chao1,
    Estimate = summary_chao1$coefficients[, "Estimate"],
    StdError = summary_chao1$coefficients[, "Std. Error"],
    tValue = summary_chao1$coefficients[, "t value"],
    Pvalue = summary_chao1$coefficients[, "Pr(>|t|)"]
  )
  
  # Ergebnisse für Shannon
  terms_shannon <- rownames(summary_shannon$coefficients)
  t_shannon <- data.frame(
    Metric = "Shannon",
    Nutrient = nutrient,
    Term = terms_shannon,
    Estimate = summary_shannon$coefficients[, "Estimate"],
    StdError = summary_shannon$coefficients[, "Std. Error"],
    tValue = summary_shannon$coefficients[, "t value"],
    Pvalue = summary_shannon$coefficients[, "Pr(>|t|)"]
  )
  
  # Kombiniere Ergebnisse
  results_list[[nutrient]] <- rbind(t_chao1, t_shannon)
}

# Ergebnisse in eine Tabelle zusammenführen
final_table <- do.call(rbind, results_list)

# P-Werte anpassen (Benjamini-Hochberg)
final_table$Pvalue_adj <- p.adjust(final_table$Pvalue, method = "BH")

# Signifikanzsternchen hinzufügen direkt in Pvalue
final_table$Pvalue <- paste0(
  round(final_table$Pvalue, 4),
  ifelse(final_table$Pvalue < 0.001, "***",
         ifelse(final_table$Pvalue < 0.01, "**",
                ifelse(final_table$Pvalue < 0.05, "*", "")))
)

# Signifikanzsternchen hinzufügen direkt in Pvalue_adj
final_table$Pvalue_adj <- paste0(
  round(as.numeric(gsub("\\*", "", final_table$Pvalue_adj)), 4),
  ifelse(as.numeric(gsub("\\*", "", final_table$Pvalue_adj)) < 0.001, "***",
         ifelse(as.numeric(gsub("\\*", "", final_table$Pvalue_adj)) < 0.01, "**",
                ifelse(as.numeric(gsub("\\*", "", final_table$Pvalue_adj)) < 0.05, "*", "")))
)

# Ergebnisse anzeigen
final_table

```



```{r}
# Schleife für alle Nährstoffe
for (nutrient in nutrients) {
  # Chao1-Modell
  lm_chao1 <- lm(as.formula(paste("Value ~", nutrient, "+ BMI + Geschlecht + Alter + Rauchen.")), data = df55)
  summary_chao1 <- summary(lm_chao1)
  
  # Shannon-Modell
  lm_shannon <- lm(as.formula(paste("Value ~", nutrient, "+ BMI + Geschlecht + Alter + Rauchen.")), data = df66)
  summary_shannon <- summary(lm_shannon)
  
  # Ergebnisse für Chao1 extrahieren
  terms_chao1 <- rownames(summary_chao1$coefficients)
  t_chao1 <- data.frame(
    Metric = "Chao1",
    Nutrient = nutrient,
    Term = terms_chao1,
    Estimate = summary_chao1$coefficients[, "Estimate"],
    StdError = summary_chao1$coefficients[, "Std. Error"],
    tValue = summary_chao1$coefficients[, "t value"],
    Pvalue = summary_chao1$coefficients[, "Pr(>|t|)"]
  )
  
  # Ergebnisse für Shannon extrahieren
  terms_shannon <- rownames(summary_shannon$coefficients)
  t_shannon <- data.frame(
    Metric = "Shannon",
    Nutrient = nutrient,
    Term = terms_shannon,
    Estimate = summary_shannon$coefficients[, "Estimate"],
    StdError = summary_shannon$coefficients[, "Std. Error"],
    tValue = summary_shannon$coefficients[, "t value"],
    Pvalue = summary_shannon$coefficients[, "Pr(>|t|)"]
  )
  
  # Kombiniere Ergebnisse für beide Metriken
  results_list[[nutrient]] <- rbind(t_chao1, t_shannon)
}

# Ergebnisse in eine Tabelle zusammenführen
final_table <- do.call(rbind, results_list)

# P-Werte anpassen (Benjamini-Hochberg)
final_table$Pvalue_adj <- p.adjust(final_table$Pvalue, method = "BH")

# Sternchen hinzufügen direkt in Pvalue
final_table$Pvalue <- paste0(
  round(final_table$Pvalue, 4),
  ifelse(final_table$Pvalue < 0.001, "***",
         ifelse(final_table$Pvalue < 0.01, "**",
                ifelse(final_table$Pvalue < 0.05, "*", "")))
)

# Sternchen hinzufügen direkt in Pvalue_adj
final_table$Pvalue_adj <- paste0(
  round(as.numeric(gsub("\\*", "", final_table$Pvalue_adj)), 4),
  ifelse(as.numeric(gsub("\\*", "", final_table$Pvalue_adj)) < 0.001, "***",
         ifelse(as.numeric(gsub("\\*", "", final_table$Pvalue_adj)) < 0.01, "**",
                ifelse(as.numeric(gsub("\\*", "", final_table$Pvalue_adj)) < 0.05, "*", "")))
)

# Zeige die Tabelle an
final_table
```



# p-value adjust with BH

#protein
```{r}
# Füge eine Spalte für die angepassten p-Werte hinzu
lm_results_protein <- data.frame(
  Variable = c("Protein", "BMI", "Geschlecht", "Alter", "Rauchen"),
  Estimate = c(0.469, -0.445, 2.525, 0.291, -2.251),
  PValue = c(0.33, 0.48, 0.65, 0.22, 0.81)
)

# Benjamini-Hochberg-Korrektur anwenden
lm_results_protein$AdjustedPValue <- p.adjust(lm_results_protein$PValue, method = "BH")

# Ausgabe der Ergebnisse
print(lm_results_protein)
```


#magnesium
```{r}
# Füge eine Spalte für die angepassten p-Werte hinzu
lm_results <- data.frame(
  Variable = c("Magnesium", "BMI", "Geschlecht", "Alter", "Rauchen"),
  Estimate = c(-0.04128, -0.00520, -0.04774, 0.00347, -0.02944),
  PValue = c(0.041, 0.311, 0.297, 0.080, 0.701)
)

# Benjamini-Hochberg-Korrektur anwenden
lm_results$AdjustedPValue <- p.adjust(lm_results$PValue, method = "BH")

# Ausgabe der Ergebnisse
print(lm_results)
```

# Session info