---
title: "Merge files into phyloseq"
author: "Mona Hildebrandt"
date: "`r Sys.Date()`"
output:
   #BiocStyle::html_document:
   html_document:
     
     code_folding: hide
editor_options:
  chunk_output_type: console
params:
  FIGPATH: "figures/"
  d.out: "./"
---

# Preparations
## Set global options

```{r style, results="asis", cache=FALSE, message = F}
# Set knit global options
library("knitr")
options(digits = 2, width = 80)
golden_ratio <- (1 + sqrt(5)) / 2
opts_chunk$set(echo = TRUE,
               tidy = FALSE,
               include = TRUE,
               fig.path = params$FIGPATH,
               dev=c("png",'pdf'),
               fig.height = 5,
               fig.width = 4 * golden_ratio,
               comment = '  ',
               dpi = 300,
               cache = FALSE)


# Pretty outputs
library("rmarkdown")
#library("BiocStyle")
library("ggthemes") # add ggplot2 themes
library("ggplot2")
library("ggpubr")

# Set seed for reproducibility
set.seed(100)

# Set plotting theme
theme_set(theme_few(base_size = 14))

# Set output directory
d.out <- params$d.out
rm(params)
```


## Load libraries for the session
```{r libraries, message=FALSE}
library("magrittr")
library("ggplot2")
library("tidyverse")
library("readxl")
#library("dada2")
library("phyloseq")
library("gridExtra")
library("vegan")
library("plyr")
library("scales")
library("reshape2")
#library("DESeq2")
library("dplyr") #for data manipulation
#library("msa") #multiple sequence alignment
library("ape") #to create phylogenetic tree
library("randomForest") 
library("caret") 
library("broom")
library("mlbench")
library("plotROC")

```


# Beta Diversity

## No Filter taxa

### Import data


Filtered phyloseq, stool samples
I have filtered here only ASV with prevalence > 5% of all samples
```{r}
# Phyloseq object AD filtered samples
d.ps.filter.ad.st <- "~/Documents/dietproject"
ps.filter.ad.st <- "/3.ps.filter.pd.abundance.rds" %>%
  paste0(d.ps.filter.ad.st,"/", .) %>% 
  readRDS() 
ps.filter.ad.st


```

subset only stool samples:

```{r}
ps.st <- subset_samples(ps.filter.ad.st, !(sample_names(ps.filter.ad.st) %in% "23Jul94-DL185"))
ps.st
```





# Beta diversity

  * How different is the microbial composition in one environment compared to another?
  * We compute the dissimilarity matrix using one of the appropriate distances for microbiome data: Bray-Curtis.
    
  

## Distance matrix
```{r}
#Bray-Curtis for ST samples:
bray_dist = phyloseq::distance(ps.st, method="bray", weighted=T)
```

```{r, results='hide'}
o.outlier <- outlier(as.matrix(bray_dist),thresh=-2.8,y=0.5)
o.outlier                                    
plot(o.outlier) 

##Sample "J31621" is outlier, remove it
#21Jun164-DL038, J31621 , I20206, 21May3544-DL370, J31637 

```

# Ordination
Measures the similarity of the samples and projects this measure into 2-dimensions.

## NMDS 
**Stress**: value representing the difference between distance in the reduced dimension compared to the complete multidimensional space.

**NMDS**: tries to minimize the stress value using gradient decent.


```{r, out.width="60%" , warning=FALSE, message=FALSE}

# Plot NMDS: using bray distances by defult. 
ordination2 <- ordinate(ps.st, method = "NMDS" )
ordination2
```


```{r}
#Add centroids
NMDS1 <- ordination2$points[,1]
NMDS2 <- ordination2$points[,2]

```

```{r}
d <- data.frame(sample_data(ps.st))
```

#Macronutrients 

#Protein
```{r}
Protein <- sample_data(ps.st)$Protein.in..
df1 <- data.frame(Protein, NMDS1, NMDS2)
```

```{r}
# set the plot with colour 
p1 <- ggplot(df1, aes(NMDS1, NMDS2))
p1 + geom_point(aes(colour = Protein))+ggtitle("Protein")+
  scale_colour_gradient2()
```

#Fat 
```{r}
Fat <- sample_data(ps.st)$Fett.in..
df2 <- data.frame(Fat, NMDS1, NMDS2)
```

```{r}
# set the plot with colour 
p2 <- ggplot(df2, aes(NMDS1, NMDS2))
p2 + geom_point(aes(colour = Fat))+ggtitle("Fat")+
  scale_colour_gradient2()
```

#Fat_MUFA
```{r}
Fat_MUFA <- sample_data(ps.st)$Fett_MUFS.in..
df3 <- data.frame(Fat_MUFA, NMDS1, NMDS2)
```

```{r}
# set the plot with colour 
p3 <- ggplot(df3, aes(NMDS1, NMDS2))
p3 + geom_point(aes(colour = Fat_MUFA))+ggtitle("Fat_MUFA")+
  scale_colour_gradient2()
```

#Carbohydrate
```{r}
Carbohydrate <- sample_data(ps.st)$Kohlenhydrate.in..
df4 <- data.frame(Carbohydrate, NMDS1, NMDS2)
```

```{r}
# set the plot with colour 
p4 <- ggplot(df4, aes(NMDS1, NMDS2))
p4 + geom_point(aes(colour = Carbohydrate))+ggtitle("Carbohydrate")+
  scale_colour_gradient2()
```

#sugar
```{r}
Sugar <- sample_data(ps.st)$davon_Zucker.in..
df5 <- data.frame(Sugar, NMDS1, NMDS2)
```

```{r}
# set the plot with colour 
p5 <- ggplot(df5, aes(NMDS1, NMDS2))
p5 + geom_point(aes(colour = Sugar))+ggtitle("Sugar")+
  scale_colour_gradient2()
```

#fiber
```{r}
Fiber <- sample_data(ps.st)$Ballaststoffe.in..
df6 <- data.frame(Fiber, NMDS1, NMDS2)
```

```{r}
# set the plot with colour 
p6 <- ggplot(df6, aes(NMDS1, NMDS2))
p6 + geom_point(aes(colour = Fiber))+ggtitle("Fiber")+
  scale_colour_gradient2()
```


#micronutrients

#cholesterol 
```{r}
Cholesterol <- sample_data(ps.st)$Cholesterin 
df7 <- data.frame(Cholesterol, NMDS1, NMDS2)
```

```{r}
# set the plot with colour 
p7 <- ggplot(df7, aes(NMDS1, NMDS2))
p7 + geom_point(aes(colour = Cholesterol))+ggtitle("Cholesterol")+
  scale_colour_gradient2()
```

#Sodium 
```{r}
Sodium <- sample_data(ps.st)$Natrium 
df8 <- data.frame(Cholesterol, NMDS1, NMDS2)
```

```{r}
# set the plot with colour 
p8 <- ggplot(df8, aes(NMDS1, NMDS2))
p8 + geom_point(aes(colour = Sodium))+ggtitle("Sodium")+
  scale_colour_gradient2()
```

#Potassium 
```{r}
Potassium <- sample_data(ps.st)$Kalium
df9 <- data.frame(Potassium, NMDS1, NMDS2)
```

```{r}
# set the plot with colour 
p9 <- ggplot(df9, aes(NMDS1, NMDS2))
p9 + geom_point(aes(colour = Potassium))+ggtitle("Potassium")+
  scale_colour_gradient2()
```

#Calcium
```{r}
Calcium <- sample_data(ps.st)$Calcium
df10 <- data.frame(Calcium, NMDS1, NMDS2)
```

```{r}
# set the plot with colour 
p10 <- ggplot(df10, aes(NMDS1, NMDS2))
p10 + geom_point(aes(colour = Calcium))+ggtitle("Calcium")+
  scale_colour_gradient2()
```

#Phosphorus
```{r}
Phosphorus <- sample_data(ps.st)$Phosphor
df11 <- data.frame(Phosphorus, NMDS1, NMDS2)
```

```{r}
# set the plot with colour 
p11 <- ggplot(df11, aes(NMDS1, NMDS2))
p11 + geom_point(aes(colour = Calcium))+ggtitle("Phosphorus")+
  scale_colour_gradient2()
```

#Magnesium
```{r}
Magnesium <- sample_data(ps.st)$Magnesium
df12 <- data.frame(Magnesium, NMDS1, NMDS2)
```

```{r}
# set the plot with colour 
p12 <- ggplot(df12, aes(NMDS1, NMDS2))
p12 + geom_point(aes(colour = Magnesium))+ggtitle("Magnesium")+
  scale_colour_gradient2()
```

#Iron
```{r}
Iron <- sample_data(ps.st)$Eisen
df13 <- data.frame(Iron, NMDS1, NMDS2)
```

```{r}
# set the plot with colour 
p13 <- ggplot(df13, aes(NMDS1, NMDS2))
p13 + geom_point(aes(colour = Iron))+ggtitle("Iron")+
  scale_colour_gradient2()
```


#Test


## Multivariate: Permanova

**Aim**: test of differences in microbial composition between 2 or more groups of samples.

**Method**: PERMANOVA

**Hypothesis testing**: centroids of Group clusters differ.

**Statistics**: test for simmilarity based on a chosen distance (here Bray-Curtis).


Separate groups

```{r}
#Bray-Curtis for ST samples:
bray1 = phyloseq::distance(ps.st, method="bray", weighted=T)

```

##Macronutrients
#Protein
```{r}
permanova1 <- adonis2(bray1 ~ sample_data(ps.st)$Protein.in..)
permanova1
```

```{r}
#tables
t1 <- data.frame(Test = "Adonis",
                 Nutrient="Protein",
                 Distance = "Bray",
                 Df = permanova1$Df,
                 SumOfSqs = permanova1$SumOfSqs,
                 R2= permanova1$R2,
                 F=permanova1$F,
                 Pvalue=permanova1$`Pr(>F)`)
```

#Fat
```{r}
permanova2 <- adonis2(bray1 ~ sample_data(ps.st)$Fett.in..)
permanova2
```

```{r}
#tables
t2 <- data.frame(Test = "Adonis",
                 Nutrient="Fat",
                 Distance = "Bray",
                 Df = permanova2$Df,
                 SumOfSqs = permanova2$SumOfSqs,
                 R2= permanova2$R2,
                 F=permanova2$F,
                 Pvalue=permanova2$`Pr(>F)`)
```

#Fat_MUFA
```{r}
permanova3 <- adonis2(bray1 ~ sample_data(ps.st)$Fett_MUFS.in..)
permanova3
```

```{r}
#tables
t3 <- data.frame(Test = "Adonis",
                 Nutrient="Fat_MUFA",
                 Distance = "Bray",
                 Df = permanova3$Df,
                 SumOfSqs = permanova3$SumOfSqs,
                 R2= permanova3$R2,
                 F=permanova3$F,
                 Pvalue=permanova3$`Pr(>F)`)
```

#Carbohydrate
```{r}
permanova4 <- adonis2(bray1 ~ sample_data(ps.st)$Kohlenhydrate.in..)
permanova4
```

```{r}
#tables
t4 <- data.frame(Test = "Adonis",
                 Nutrient="Carbohydrate",
                 Distance = "Bray",
                 Df = permanova4$Df,
                 SumOfSqs = permanova4$SumOfSqs,
                 R2= permanova4$R2,
                 F=permanova4$F,
                 Pvalue=permanova4$`Pr(>F)`)
```

#Sugar
```{r}
permanova5 <- adonis2(bray1 ~ sample_data(ps.st)$davon_Zucker.in..)
permanova5
```

```{r}
#tables
t5 <- data.frame(Test = "Adonis",
                 Nutrient="Sugar",
                 Distance = "Bray",
                 Df = permanova5$Df,
                 SumOfSqs = permanova5$SumOfSqs,
                 R2= permanova5$R2,
                 F=permanova5$F,
                 Pvalue=permanova5$`Pr(>F)`)
```

#Fiber
```{r}
permanova6 <- adonis2(bray1 ~ sample_data(ps.st)$Ballaststoffe.in..)
permanova6
```

```{r}
#tables
t6 <- data.frame(Test = "Adonis",
                 Nutrient="Fiber",
                 Distance = "Bray",
                 Df = permanova6$Df,
                 SumOfSqs = permanova6$SumOfSqs,
                 R2= permanova6$R2,
                 F=permanova6$F,
                 Pvalue=permanova6$`Pr(>F)`)
```

##Micronutrients
#Cholesterol
```{r}
permanova7 <- adonis2(bray1 ~ sample_data(ps.st)$Cholesterin)
permanova7
```

```{r}
#tables
t7 <- data.frame(Test = "Adonis",
                 Nutrient="Cholesterol",
                 Distance = "Bray",
                 Df = permanova7$Df,
                 SumOfSqs = permanova7$SumOfSqs,
                 R2= permanova7$R2,
                 F=permanova7$F,
                 Pvalue=permanova7$`Pr(>F)`)
```

#Sodium
```{r}
permanova8 <- adonis2(bray1 ~ sample_data(ps.st)$Natrium)
permanova8
```

```{r}
#tables
t8 <- data.frame(Test = "Adonis",
                 Nutrient="Sodium",
                 Distance = "Bray",
                 Df = permanova8$Df,
                 SumOfSqs = permanova8$SumOfSqs,
                 R2= permanova8$R2,
                 F=permanova8$F,
                 Pvalue=permanova8$`Pr(>F)`)
```

#Potassium
```{r}
permanova9 <- adonis2(bray1 ~ sample_data(ps.st)$Kalium)
permanova9
```

```{r}
#tables
t9 <- data.frame(Test = "Adonis",
                 Nutrient="Potassium",
                 Distance = "Bray",
                 Df = permanova9$Df,
                 SumOfSqs = permanova9$SumOfSqs,
                 R2= permanova9$R2,
                 F=permanova9$F,
                 Pvalue=permanova9$`Pr(>F)`)
```

#Calcium
```{r}
permanova10 <- adonis2(bray1 ~ sample_data(ps.st)$Calcium)
permanova10
```

```{r}
#tables
t10 <- data.frame(Test = "Adonis",
                 Nutrient="Calcium",
                 Distance = "Bray",
                 Df = permanova10$Df,
                 SumOfSqs = permanova10$SumOfSqs,
                 R2= permanova10$R2,
                 F=permanova10$F,
                 Pvalue=permanova10$`Pr(>F)`)
```

#Phosphorus
```{r}
permanova11 <- adonis2(bray1 ~ sample_data(ps.st)$Phosphor)
permanova11
```

```{r}
#tables
t11 <- data.frame(Test = "Adonis",
                 Nutrient="Phosphorus",
                 Distance = "Bray",
                 Df = permanova11$Df,
                 SumOfSqs = permanova11$SumOfSqs,
                 R2= permanova11$R2,
                 F=permanova11$F,
                 Pvalue=permanova11$`Pr(>F)`)
```

#Magnesium
```{r}
permanova12 <- adonis2(bray1 ~ sample_data(ps.st)$Magnesium)
permanova12
```

```{r}
#tables
t12 <- data.frame(Test = "Adonis",
                 Nutrient="Magnesium",
                 Distance = "Bray",
                 Df = permanova12$Df,
                 SumOfSqs = permanova12$SumOfSqs,
                 R2= permanova12$R2,
                 F=permanova12$F,
                 Pvalue=permanova12$`Pr(>F)`)
```

#Iron 
```{r}
permanova13 <- adonis2(bray1 ~ sample_data(ps.st)$Eisen)
permanova13
```

```{r}
#tables
t13 <- data.frame(Test = "Adonis",
                 Nutrient="Iron",
                 Distance = "Bray",
                 Df = permanova13$Df,
                 SumOfSqs = permanova13$SumOfSqs,
                 R2= permanova13$R2,
                 F=permanova13$F,
                 Pvalue=permanova13$`Pr(>F)`)
```





